addons/sale_timesheet/views/project_task_views.xml

```diff
@@ -25,13 +25,18 @@
                             <field name="sale_line_id" column_invisible="not parent.id" required="True" groups="!sales_team.group_sale_salesman" options="{'no_create': True}"
                                 context="{'search_default_order_id': sale_order_id}"/>
                             <field name="sale_line_id" column_invisible="not parent.id" required="True" groups="sales_team.group_sale_salesman"
+                                options="{'no_create': True, 'no_open': True}"
```

Comment:
I don't understand why you added the 'no_open' here. Since this field will be filled with the default value set on the project, not being allowed to open it means the user is locked with that default value.
After checking on a localhost, it looks like the 'no_open' is not even working correctly anyway. I was still able to remove the sol, and then make the create button appears. (the creation & updates worked well though).
Last comment:
From a functional point of view, its not very intuitive. You have to manually remove the default value to be able to see the button to create a new sol... I feel like only the users that know that this feature exists will be able to use it. I'd suggest we show the button even if a value is set